File: app/src/main/java/com/example/Maatran/ui/WelcomeActivity.java
package com.example.Maatran.ui;

import android.content.Intent;
import android.os.Bundle;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.content.res.AppCompatResources;

import com.example.Maatran.R;
import com.example.Maatran.tests.AppNavigationActivity;
import com.google.android.material.button.MaterialButton;

//main Activity
//XML file: home_1
public class WelcomeActivity extends AppCompatActivity {

private static final String TAG="WelcomeActivity";
MaterialButton signInBtn;
MaterialButton signUpBtn;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_welcome_screen);
        getWindow().setStatusBarColor(getResources().getColor(R.color.welcome_accent));
        Button test_btn = findViewById(R.id.test_btn);
        test_btn.setVisibility(View.INVISIBLE);
        test_btn.setOnClickListener(v-> signInWithTestAccount());
        signInBtn = findViewById(R.id.welcome_signin_btn);
        signUpBtn = findViewById(R.id.welcome_register_btn);
    }

    @Override
    protected void onStart() {
        super.onStart();
        signInBtn.setOnClickListener(v->signInOptions());
        signInBtn.setOnTouchListener((v, event) -> {
            if(event.getAction()==MotionEvent.ACTION_DOWN)
            {
                v.setBackground(AppCompatResources.getDrawable(this,R.drawable.welcome_button_clickedstate));
                changeTextColor(1,signInBtn);
                return true;
            }
            if(event.getAction()==MotionEvent.ACTION_UP)
            {
                v.performClick();
                v.setBackground(AppCompatResources.getDrawable(this,R.drawable.welcome_button_nonclickedstate));
                changeTextColor(0,signInBtn);
                return true;
            }
            return false;
        });
        signUpBtn.setOnClickListener(v->registerOptions());
        signUpBtn.setOnTouchListener((v, event) -> {
            if(event.getAction()==MotionEvent.ACTION_DOWN)
            {
                v.setBackground(AppCompatResources.getDrawable(this,R.drawable.welcome_button_clickedstate));
                changeTextColor(1,signUpBtn);
                return true;
            }
            if(event.getAction()==MotionEvent.ACTION_UP)
            {
                v.performClick();
                v.setBackground(AppCompatResources.getDrawable(this,R.drawable.welcome_button_nonclickedstate));
                changeTextColor(0,signUpBtn);
                return true;
            }
            return false;
        });
    }

    public void changeTextColor(int type, Button btn)
    {
        if(type==1)
            btn.setTextColor(getResources().getColor(R.color.white));
        else
            btn.setTextColor(getResources().getColor(R.color.welcome_accent));
    }

    //for signing-in, calls LoginActivity.class
    public void signInOptions() {
        Intent intent = new Intent(getApplicationContext(), LoginActivity.class);
        startActivity(intent);
    }

    //for registration, calls RegistrationActivity.class
    public void registerOptions() {
        Intent intent  = new Intent(getApplicationContext(), SignUpActivity.class);
        startActivity(intent);
    }

    private void signInWithTestAccount()
    {
          Intent intent = new Intent(this, AppNavigationActivity.class);
          startActivity(intent);
//        String email = "test1@gmail.com";
//        String password = "123456";
//        FirebaseAuth auth = FirebaseAuth.getInstance();
//        auth.signInWithEmailAndPassword(email, password)
//                .addOnCompleteListener(this, task -> {
//                    if (task.isSuccessful()) {
//                        // Sign in success, update UI with the signed-in user's information
//                        Log.d(TAG, "signInWithEmail:success");
//                        Toast.makeText(this,"WARNING: USE THIS FEATURE FOR DEBUG/TEST PURPOSES ONLY",Toast.LENGTH_LONG).show();
//                        Intent intent = new Intent(getApplicationContext(), DashboardActivity.class);
//                        startActivity(intent);
//                    } else {
//                        // If sign in fails, display a message to the user.
//                        Log.w(TAG, "signInWithEmail:failure", task.getException());
//                        Toast.makeText(this, "Authentication failed.",
//                                Toast.LENGTH_SHORT).show();
//                    }
//                });
    }

}

File: app/src/main/java/com/example/Maatran/tests/WelcomeActivity.kt
package com.example.Maatran.tests

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.MotionEvent
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.content.res.AppCompatResources
import com.example.Maatran.R
import com.example.Maatran.databinding.ActivityWelcomeScreenBinding
import com.example.Maatran.ui.SignUpActivity
import com.google.android.gms.tasks.Task
import com.google.android.material.button.MaterialButton
import com.google.firebase.auth.AuthResult
import com.google.firebase.auth.FirebaseAuth

class WelcomeActivity : AppCompatActivity() {

    private val _tag:String = "WelcomeActivity"
    private lateinit var _binding: ActivityWelcomeScreenBinding
    private lateinit var _signInBtn: MaterialButton
    private lateinit var _signUpBtn: MaterialButton
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        _binding = ActivityWelcomeScreenBinding.inflate(layoutInflater)
        val rootView = _binding.root
        setContentView(rootView)
        window.statusBarColor = resources.getColor(R.color.welcome_accent)
        val testBtn: Button = _binding.testBtn
        testBtn.visibility = View.VISIBLE
        testBtn.setOnClickListener {
        signInWithTestAccount()
        }
        _signInBtn = _binding.welcomeSigninBtn
        _signUpBtn = _binding.welcomeRegisterBtn
    }

    override fun onStart() {
        super.onStart()
        _signInBtn.setOnClickListener{ signInOptions()}
        _signInBtn.setOnTouchListener { v: View, event: MotionEvent ->
            if (event.action == MotionEvent.ACTION_DOWN) {
                v.background = AppCompatResources.getDrawable(
                    this,
                    R.drawable.welcome_button_clickedstate
                )
                changeTextColor(1, _signInBtn)
                return@setOnTouchListener true
            }
            if (event.action == MotionEvent.ACTION_UP) {
                v.performClick()
                v.background = AppCompatResources.getDrawable(
                    this,
                    R.drawable.welcome_button_nonclickedstate
                )
                changeTextColor(0, _signInBtn)
                return@setOnTouchListener true
            }
            false
        }
        _signUpBtn.setOnClickListener{ registerOptions()}
        _signUpBtn.setOnTouchListener{v, event ->
            if(event.action == MotionEvent.ACTION_DOWN)
            {
                v.background = AppCompatResources.getDrawable(
                    this,
                    R.drawable.welcome_button_clickedstate
                )
                changeTextColor(1, _signUpBtn)
                return@setOnTouchListener true
            }
            if(event.action == MotionEvent.ACTION_UP)
            {
                v.performClick()
                v.background = AppCompatResources.getDrawable(
                    this,
                    R.drawable.welcome_button_nonclickedstate
                )
                changeTextColor(0, _signUpBtn)
                return@setOnTouchListener true
            }
            return@setOnTouchListener false
        }
    }

    private fun changeTextColor(type: Int, btn: MaterialButton)
    {
        if(type == 1)
            btn.setTextColor(resources.getColor(R.color.white))
        else
            btn.setTextColor(resources.getColor(R.color.welcome_accent))
    }

    private fun signInOptions() {
        val intent = Intent(applicationContext, LoginActivity::class.java)
        startActivity(intent)
    }

    private fun registerOptions() {
        val intent = Intent(applicationContext, SignUpActivity::class.java)
        startActivity(intent)
    }

    private fun signInWithTestAccount() {
        val email = "test1@gmail.com"
        val password = "123456"
        val auth = FirebaseAuth.getInstance()
        auth.signInWithEmailAndPassword(email, password)
            .addOnCompleteListener(
                this
            ) { task: Task<AuthResult?> ->
                if (task.isSuccessful) {
                    // Sign in success, update UI with the signed-in user's information
                    Log.d(_tag, "signInWithEmail:success")
                    Toast.makeText(
                        this,
                        "WARNING: USE THIS FEATURE FOR DEBUG/TEST PURPOSES ONLY",
                        Toast.LENGTH_LONG
                    ).show()
                    val intent =
                        Intent(applicationContext, AppNavigationActivity::class.java)
                    startActivity(intent)
                } else {
                    // If sign in fails, display a message to the user.
                    Log.w(
                        _tag,
                        "signInWithEmail:failure",
                        task.exception
                    )
                    Toast.makeText(
                        this, "Authentication failed.",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            }
    }
}